src/FFmpeg.h

```diff
@@ -128,6 +128,13 @@ extern "C" {
       }
       #endif
    #endif
+
+   #if LIBAVCODEC_VERSION_MAJOR < 58
```

Comment:
Thank you for this catch, it's simple enough to test for existing defines and support everything.

Reply:
Section A.12.3 of my good old K & R book (ANSI C version) says that when a a macro is expanded, occurrences of the same macro name in the resulting token sequence are not re-expanded.  So this will not produce undefined results from the preprocessor.  Rather AV_CODEC_FLAG_QSCALE will expand to CODEC_FLAG_QSCALE which will expand to AV_CODEC_FLAG_QSCALE again and then it stops.

Still, I'd agree we should not rely on that, and instead invent a third constant name to use in those few places in Audacity.

PRL